### PR TITLE
Added new StringJoin() utility function

### PR DESCRIPTION
**Pull request by larsewi**
The function joins a sequence of strings with an optional separator in between the elements.

### CODE REVIEW

**Review by craigcomstock — APPROVED** [review]
the code looks so familiar to something I saw recently! ;) 🏆

File: libutils/string_lib.c [hunk 1]
```diff
@@ -1781,3 +1781,23 @@ bool StringMatchesOption(
...
     return StringEqualN_IgnoreCase(supplied, longopt, length);
 }
+
+char *StringJoin(const Seq *const seq, const char *sep)
+{
+    assert(seq != NULL);
+
+    Writer *const writer = StringWriter();
+    const size_t len = SeqLength(seq);
+    for (size_t i = 0; i < len; i++)
```
Comment (vpodzime) [inline libutils/string_lib.c:1791]: I'd do `i < (len -1)` writing the separator *after* every item and then writing the last item after the `for` loop without a separator. But YMMV.

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise.

File: libutils/string_lib.c (hunk 1 above)
Comment (vpodzime) [reply libutils/string_lib.c:1791]: Yup, but that's easy to check upfront.

**Pull request merged**